src/Service/GlobalAuthenticationService.php

```diff
@@ -1,120 +1,137 @@
 <?php
+
 namespace ID3Global\Service;
 
+use Exception;
 use ID3Global\Exceptions\IdentityVerificationFailureException;
 use ID3Global\Gateway\GlobalAuthenticationGateway;
 use ID3Global\Identity\Identity;
+use SoapClient;
+use stdClass;
 
 class GlobalAuthenticationService extends ID3BaseService
 {
     /**
-     * @var \ID3Global\Identity\Identity
+     * @var Identity
      */
-    private $identity;
+    private Identity $identity;
 
     /**
      * @var string The Profile ID to be used when verifying a @link \ID3Global\Identity\Identity object
      */
-    private $profileID;
+    private string $profileID;
 
     /**
      * @var int The Profile Version to be used when verifying a @link \ID3Global\Identity\Identity object. The version
-     * 0 represents the 'most recent version of the profile', which is generally what is required.
+     *          0 represents the 'most recent version of the profile', which is generally what is required.
      */
-    private $profileVersion;
+    private int $profileVersion = 0;
 
     /**
-     * @var string A reference stored against this identity request. This is optional, but is recommended to set a
-     * customer reference and store it against the returned identity verification so it can be later tracked if
-     * necessary for compliance purposes.
+     * @var string|null A reference stored against this identity request. This is optional, but is recommended to set a
+     *                  customer reference and store it against the returned identity verification so it can be later tracked if
+     *                  necessary for compliance purposes.
      */
-    private $customerReference;
+    private ?string $customerReference = null;
 
     /**
      * @var GlobalAuthenticationGateway
      */
-    private $gateway;
+    private GlobalAuthenticationGateway $gateway;
 
     /**
-     * @var \stdClass The last response, directly from the API gateway. Can be retrieved using
-     * {@link getLastVerifyIdentityResponse()}.
+     * @var stdClass|null The last response, directly from the API gateway. Can be retrieved using
+     *                    {@link getLastVerifyIdentityResponse()}.
      */
-    private $lastVerifyIdentityResponse = null;
-    
+    private ?stdClass $lastVerifyIdentityResponse = null;
+
     /**
-     * @var string The last request passed into SoapClient
+     * @var string|null The last request passed into SoapClient
      */
-    private $lastRawRequest = null;
+    private ?string $lastRawRequest = null;
 
     /**
      * @var array Optional extras for the SOAP client
      */
-    private $soapOptions = array();
+    private array $soapOptions = [];
+
+    public function __construct(
```

Comment:
This now forces you to specify the identity during service construction, as well as a gateway. The gateway is less of an issue - one service should generally be akin to one gateway (and that's the same in this case), however I'm less okay with passing `Identity` and `$profileId` in here.

In particular, the `IGlobalAuthenticate` endpoint port in the WSDL file defines the following functions: `AuthenticateSP`, `IncrementalVerification`, `AuthenticateMP`, `ForceOverride` and `AddressLookup`. _Theoretically_, at least in my head, this service class might reasonably implement methods for all of these (realistically I think only `AuthenticateMP` makes sense). I'd rather not tie the constructor to what is required for just one of these WSDL ports - hopefully that makes sense?

I'd prefer if we just use the setters as necessary prior to calling ->verifyIdentity(), it feels easier to extend in future that way.

Note: Adjusting this will mean the README will have to adjust too.

Happy to hear your thoughts on this, it's probably my biggest piece of feedback :)

Reply:
I agree. Initially I wanted to take off these properties and make the `verifyIdentity` method accept them as arguments, since they are not part of the state, rather inputs for the service functions. I just did not want to make much change in the class structure without consulting. I am happy to see you think that way now.